### PR TITLE
Simplify top-level wallet state

### PR DESCRIPTION
**Pull request by tomclose**
This PR removes the `initializing` and `initialized` stages, replacing them with a top-level wallet state with four different types.

The motivation here was to try and simplify things a bit, in preparation for a potential move to actions-first reducers.

### CODE REVIEW

File: packages/wallet/src/redux/state.ts [hunk 1]
```diff
@@ -1,10 +1,85 @@
...
+// -----------
+export const WAIT_FOR_LOGIN = 'INITIALIZING.WAIT_FOR_LOGIN';
+export const METAMASK_ERROR = 'INITIALIZING.METAMASK_ERROR';
+export const WAIT_FOR_ADJUDICATOR = 'INITIALIZING.WAIT_FOR_ADJUDICATOR';
+export const WALLET_INITIALIZED = 'WALLET.INITIALIZED';
 
-export type WalletState = InitializingState | InitializedState;
+// ------
+// States
+// ------
```
Comment (tomclose) [inline packages/wallet/src/redux/state.ts]: @andrewgordstewart do you think we can safely remove `channelState`, `fundingState` and `outboxState` properties from everything but the `Initialized` state?

Comment (andrewgordstewart) [reply packages/wallet/src/redux/state.ts]: Yes

File: packages/wallet/src/redux/state.ts [hunk 2]
```diff
@@ -1,10 +1,85 @@
...
+  return { type: METAMASK_ERROR, outboxState, fundingState, channelState };
+}
+
+export function waitForAdjudicator(params: Properties<WaitForAdjudicator>): WaitForAdjudicator {
+  const { outboxState, fundingState, channelState, uid } = params;
+  return { type: WAIT_FOR_ADJUDICATOR, outboxState, fundingState, channelState, uid };
+}
+
+export function initialized(params: Properties<Initialized>): Initialized {
+  const { outboxState, fundingState, channelState, uid, networkId, adjudicator } = params;
```
Comment (lalexgap) [inline packages/wallet/src/redux/state.ts]: Minor but is there a reason we can't just do this?:
 ```
return {
    type: WALLET_INITIALIZED,
    ...params
}
```

Comment (tomclose) [reply packages/wallet/src/redux/state.ts]: Yeah - `Properties<Initialized>` could also contain extra keys. (I'm not sure if that's actually important here, but it was useful when we were constructing channel states.)

**Pull request merged**